src/triage.python/worker.py

```diff
@@ -0,0 +1,375 @@
+    interpreter.HandleCommand(str('command script import %s' % PATH('analysis_path')), importScriptCommandResult)
+    if importScriptCommandResult.Succeeded():
+        VERBOSE('analysis.py imported. Running analyze.')
+        interpreter.HandleCommand(str('analyze -i %s -o ./analysis.txt' % PATH('triage_ini')), analyzeCommandResult)
+        if analyzeCommandResult.Succeeded():
+            VERBOSE('analysis command succeeded.')
+
+            with open('./analysis.txt', 'r') as analysisFile:
+                data = analysisFile.read().replace('\n', '')
+
+            VERBOSE('ANALYSIS OUTPUT: ' + analyzeCommandResult.GetOutput())
+
+            return data;
+        else:
+            FAILURE('analyze step failed with error: ' + analyzeCommandResult.GetError())
+    else:
+        FAILURE('import script failed with error: ' + importScriptCommandResult.GetError())
+
+    return '<no results>'
+
+def Analyze(testList, state):
+    UpdateWorkerState(state, 'analyzing')
+
+    if(not testList or len(testList) == 0):
+        return testList
+
+    for corePath, correlationId, jobid, testname in testList:
+        INFORMATIVE('running analyzer on ' + testname)
+        pathTuple = (corePath, correlationId, jobid, testname)
+        debuggerTuple = StartDebugger(pathTuple)
+
+        VERBOSE('We have a tuple here: ' + str(debuggerTuple))
+        result = RunAnalysis(debuggerTuple)
+        
+        Store(pathTuple, result, state)
+    
+    cleanup(debuggerTuple, state)
+
+# stores analysis results in Azure blob
+def Store(pathTuple, result, state):
+    #unpack our tuple
+    path = pathTuple[0]
+    correlationId = pathTuple[1]
+    jobId = pathTuple[2]
+    testName = pathTuple[3]
+    
+    # put stuff in to blob storage
+    # blob_service.put_blob(state, os.path.join(state, correlationId, jobId, testName), result, 'BlockBlob')
+    blob_service.create_blob_from_text(state, os.path.join(state, correlationId, jobId, testName), result)
+    VERBOSE("Getting container metadata.")
+
+    metadata = blob_service.get_container_metadata(state)
+    if metadata.has_key('job_count_completed'):
+        countCompleted = int(metadata['job_count_completed'])
+        metadata['job_count_completed'] = str(countCompleted + 1)
+    else:
+        metadata['job_count_completed'] = '1'
+    
+        
+    print str(metadata)
+    blob_service.set_container_metadata(state, metadata)
+    metadata = blob_service.get_container_metadata(state)
+    VERBOSE(str(metadata))
+
+
+import shutil
+def cleanup(debuggerTuple, state):
+    # unpack our tuple
+    debugger = debuggerTuple[0]
+    interpreter = debuggerTuple[1]
+    target = debuggerTuple[2]
+    process = debuggerTuple[3]
+    pathTuple = debuggerTuple[4]
+
+    # just delete the /home/DotNet folder for now.
+    VERBOSE('cleaning up.')
+    lldb.SBDebugger.Destroy(debugger)
+    shutil.rmtree('/home/DotNetBot/') # This will likely not work in the future when we receive dumps from other people.
```

Comment:
We can't be guaranteed that dump zips will unpack to here (as per you're comment.